### PR TITLE
[WIP] Add stopOnExit setting

### PR DESCRIPTION
**Pull request by jdmota**
For now I added this setting. It should be easier to test with it, since we can turn it on or off.

If you prefer, you can create a new branch, and I point this PR to it.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -22,6 +22,10 @@ function lookupFlowDir(context) {
     : root;
 }
 
+function stopOnExit(context) {
+  return !!(context.settings && context.settings.stopOnExit);
```
Comment (amilajack) [inline src/index.js]: How about just `context.settings && context.settings.stopOnExit`?

File: src/collect.js [hunk 2]
```diff
@@ -96,7 +96,9 @@ function getFlowBin() {
   return process.env.FLOW_BIN || flowBin;
 }
 
-function executeFlow(stdin, root, filepath) {
+let didExecute = false;
+
+function executeFlow(stdin, root, stopOnExit, filepath) {
```
Comment (amilajack) [inline src/collect.js]: Let's annotate the types of function params and all other newly created functions added in this PR

**Comment by amilajack** [comment]
What do you think about adding a timeout feature using [pm2](https://github.com/Unitech/pm2)? We can create a global list of all the servers started and timestamps:

```js
// /user/John/flowtype-error-serers.json
{
  servers: [
    {
       directory: '/users/John/project-1',
       lastRunTimestap: '3-hours-ago' // this should be a unix timestamp or something of the sort
    },
    ...
  ]
}
```

Ideally, the pm2 process would kill any flow server that hasn't been in the last 30 minutes or so.

Thoughts on this?

**Comment by jdmota** [comment]
I never worked with pm2, but it seems to be a big dependency. But I'm not sure.

File: src/index.js (hunk 1 above)
Comment (jdmota) [inline src/index.js]: I think that if I add annotations, this method should return a boolean. So using `!!` would make sense to me.

**Comment by amilajack** [comment]
@jdmota Thanks again!

**Comment by jdmota** [comment]
I still need to confirm that with this setting, the flow server is actually stopped. I will test when I get some time.

**Pull request merged**